scripts/ilo-info.nse

```diff
@@ -0,0 +1,123 @@
+description = [[
+Extracts information about HP iLO boards.
+
+HP iLO boards have an unauthenticated info disclosure at <ip>/xmldata?item=all.
+It lists board informations such as server model, firmware version, 
+MAC addresses, IP addresses etc. This script uses the slaxml library
+to parse the iLO xml file and display the info.
+]]
+
+---
+--@usage
+--nmap --script ilo-info -p 80 <host>
+--
+--@output
+--PORT   STATE SERVICE
+--80/tcp open  http
+--| ilo-info:
+--|   ILOType     : Integrated Lights-Out 4 (iLO 4)
+--|   Serial No   : ILOXXXXXXXXXX
+--|   ILOFirmware : X.XX
+--|   UUID        : XXXXXXXXXXXXXXXX
+--|   cUUID       : XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX
+--|   ServerType  : ProLiant MicroServer Gen8
+--|   ProductID   : XXXXXX-XXX
+--|   NICs:
+--|     NIC 1:
+--|       Mac Address : 12:34:56:78:9a:bc
+--|       Description : iLO 4
+--|       IP Address  : 10.10.10.10
+--|       Status      : OK
+--|     NIC 2:
+--|       Mac Address : 11:22:33:44:55:66
+--|       Description : iLO 4
+--|       IP Address  : Unknown
+--|_      Status      : Disabled
+---
+
+author = "Rajeev R Menon"
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {"safe","discovery"}
+
+local http = require "http"
+local slaxml = require "slaxml"
+local stdnse = require "stdnse"
+
+portrule = function(host,port)
```

Comment:
You should be able to use the [`shortport.http`](https://nmap.org/nsedoc/lib/shortport.html#http) portrule for this script.